OPHD/States/MapViewState.cpp

```diff
@@ -993,9 +993,12 @@ void MapViewState::populateRobotMenu()
 {
 	mRobots.clear();
 
-	for (auto item : RobotMetaTable)
+	for (auto& item : RobotMetaTable)
 	{
-
+		if (mRobotPool.robotAvailable(item.first))
+		{
+			mRobots.addItemSorted(item.second.name, item.second.sheetIndex, static_cast<int>(item.first));
```

Comment:
I recommend using structured bindings to give better names than `first` and `second`:
```cpp
for (auto& [robotType, robotMeta] : RobotMetaTable)
{
	if (mRobotPool.robotAvailable(robotType))
	{
		mRobots.addItemSorted(robotMeta.name, robotMeta.sheetIndex, static_cast<int>(robotType));
	}
}
```

Reply:
Good point, I keep forgetting I can do that. Will make a commit to address that now.